### PR TITLE
Implement compare API.

### LINKED ISSUE #10323: Implement files/compare API
Follow-up from #10322 which should implement the API related to comparing two different versions of the same add-on.

### PR DESCRIPTION
**Pull request by EnTeQuAk**
This implements fetching a diff by pathspec, we need to implement this
upstream so it's implemented as a workaround for now.

This removes support for copy and rename finding for now. I filed #10849
to fix this properly in a separate step.

The actual diff output format is hard-coded and not yet implemented as
serializers for now which makes experimenting much easier in case we
need to adapt the format depending on how the client implements it.

Fixes #10323

### CODE REVIEW

**Comment by EnTeQuAk** [comment]
I'm trying to figure out that travis failure - https://travis-ci.org/mozilla/addons-server/jobs/502683722#L2981-L2986 - which works perfectly fine for me locally.

Should be ready for a first look though.

File: src/olympia/reviewers/tests/test_views.py [hunk 1]
```diff
@@ -5289,6 +5233,70 @@ def test_unlisted_version_user_but_not_author(self):
...
+    client_class = APITestClient
+    __test__ = True
+
+    def setUp(self):
+        super(TestReviewAddonVersionViewSetDetail, self).setUp()
+
+        # TODO: Most of the initial setup could be moved to
+        # setUpTestData but unfortunately paths are setup in pytest via a
+        # regular autouse fixture that has function-scope so functions in
+        # setUpTestData don't use proper paths (cgrebs)
```
Comment (diox) [inline src/olympia/reviewers/tests/test_views.py]:
```suggestion
        # setUpTestData doesn't use proper paths (cgrebs)
```

File: docs/topics/api/reviewers.rst [hunk 2]
```diff
@@ -139,3 +141,116 @@ This endpoint allows you to browse through the contents of an Add-on version.
...
+
+This endpoint allows you to compare two Add-on versions with each other.
+
+    .. note::
+        Requires authentication and the current user to have ``ReviewerTools:View``
+        permission for listed add-ons as well as ``Addons:ReviewUnlisted`` for
+        unlisted add-ons. Additionally the current user can also be the owner
+        of the add-on.
+
+.. http:get:: /api/v4/reviewers/addon/(int:addon_id)/versions/(int:version_id)/compare/(int:version_id)/
```
Comment (willdurand) [inline docs/topics/api/reviewers.rst]: This URL is not exposed actually. It should be `compare_to` and  not `compare`.

Comment (EnTeQuAk) [reply docs/topics/api/reviewers.rst]: Indeed, fixed in 7a84c2597f

**Pull request merged**